analysis/deathknightblood/src/modules/items/CrimsonRuneWeapon.tsx

```diff
@@ -0,0 +1,97 @@
+import { formatPercentage, formatNumber } from 'common/format';
+import SPELLS from 'common/SPELLS';
+import RESOURCE_TYPES from 'game/RESOURCE_TYPES';
+import { SpellLink } from 'interface';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { ResourceChangeEvent } from 'parser/core/Events';
+import { ThresholdStyle, When } from 'parser/core/ParseResults';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import BoneShieldTimesByStacks from '../features/BoneShieldTimesByStacks';
+import UptimeIcon from 'interface/icons/Uptime';
+
+class CrimsonRuneWeapon extends Analyzer {
+  static dependencies = {
+    boneShieldTimesByStacks: BoneShieldTimesByStacks,
+  };
+
+  constructor(options: Options) {
+    super(options);
+
+    const active = this.selectedCombatant.hasLegendaryByBonusID(SPELLS.CRIMSON_RUNE_WEAPON.bonusID);
```

Comment:
this will need to change after merging `shadowlands` into your branch.
```suggestion
    const active = this.selectedCombatant.hasLegendary(SPELLS.CRIMSON_RUNE_WEAPON);
```